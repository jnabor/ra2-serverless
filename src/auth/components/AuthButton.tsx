import React from 'react'

import Button from '@material-ui/core/Button'
import { useStyles } from './styles'

export interface AuthButtonProps {
  children: any
  disabled: boolean
}

const AuthButton: React.SFC<AuthButtonProps> = ({ children, disabled }) => {
  const classes = useStyles()

  return (
    <Button
      type='submit'
      disabled={disabled}
      fullWidth
      variant='contained'
      color='primary'
      size='large'
      className={classes.submit}>
      {children}
    </Button>
  )
}

export default AuthButton
