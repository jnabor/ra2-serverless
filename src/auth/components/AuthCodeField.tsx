import React, { useState } from 'react'
import TextField from '@material-ui/core/TextField'

export interface AuthEmailFieldProps {
  setCode: (code: string) => void
}

const AuthEmailField: React.SFC<AuthEmailFieldProps> = ({ setCode }) => {
  const [hint, setHint] = useState<string>('')

  let delay: any = null
  const validate = (code: string): any => {
    if (delay !== null) {
      clearTimeout(delay)
    }
    delay = setTimeout(() => {
      const isValid = code.length === 6
      setHint(isValid ? '' : 'Incorrect code length.')
      setCode(isValid ? code : '')
      delay = null
    }, 300)
  }

  return (
    <TextField
      variant='outlined'
      margin='normal'
      required
      fullWidth
      id='confirmation'
      label='Confirmation Code'
      name='confirmation'
      helperText={hint}
      onChange={e => validate(e.currentTarget.value)}
    />
  )
}

export default AuthEmailField
