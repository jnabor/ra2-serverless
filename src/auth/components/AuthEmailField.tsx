import React, { useState } from 'react'
import TextField from '@material-ui/core/TextField'

export interface AuthEmailFieldProps {
  setEmail: (email: string) => void
}

const AuthEmailField: React.SFC<AuthEmailFieldProps> = ({ setEmail }) => {
  const [valid, setValid] = useState<boolean>(true)
  const [hint, setHint] = useState<string>('')

  let delay: any = null
  const validate = (email: string): any => {
    if (delay !== null) {
      clearTimeout(delay)
    }
    delay = setTimeout(() => {
      const isValid = /^[a-zA-Z0-9.]+@[a-zA-Z0-9]+\.[A-Za-z]+$/.test(email)
      setValid(isValid)
      setHint(
        isValid ? '' : email.length > 0 ? 'Invalid email.' : 'Enter email.'
      )
      setEmail(isValid ? email : '')
      delay = null
    }, 200)
  }

  return (
    <TextField
      error={!valid}
      variant='outlined'
      margin='normal'
      required
      fullWidth
      id='email'
      label='Email Address'
      name='email'
      autoComplete='email'
      helperText={hint}
      onChange={e => validate(e.currentTarget.value)}
    />
  )
}

export default AuthEmailField
