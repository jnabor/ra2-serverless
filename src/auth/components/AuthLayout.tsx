import React from 'react'

import Avatar from '@material-ui/core/Avatar'
import CssBaseline from '@material-ui/core/CssBaseline'
import LockOutlinedIcon from '@material-ui/icons/LockOutlined'
import Typography from '@material-ui/core/Typography'
import Container from '@material-ui/core/Container'
import { useStyles } from './styles'

export interface AuthLayoutProps {
  title: string
  children: any
}

const AuthLayout: React.SFC<AuthLayoutProps> = ({ title, children }) => {
  const classes = useStyles()

  return (
    <Container maxWidth='xs'>
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component='h1' variant='h5'>
          {title}
        </Typography>
        {children}
      </div>
    </Container>
  )
}

export default AuthLayout
