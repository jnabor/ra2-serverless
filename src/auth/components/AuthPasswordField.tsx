import React, { useState } from 'react'

import {
  FormControl,
  InputLabel,
  OutlinedInput,
  InputAdornment,
  IconButton,
  FormHelperText
} from '@material-ui/core'

import { Visibility, VisibilityOff } from '@material-ui/icons'

import {
  useTheme,
  createStyles,
  makeStyles,
  Theme
} from '@material-ui/core/styles'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    textfield: {
      margin: theme.spacing(2, 0, 0)
    }
  })
)

export interface AuthPasswordFieldProps {
  setPassword: (password: string) => void
}

const AuthPasswordField: React.SFC<AuthPasswordFieldProps> = ({
  setPassword
}) => {
  const classes = useStyles(useTheme())

  const [showPassword, setShowPassword] = useState<boolean>(false)
  const [valid, setValid] = useState<boolean>(true)
  const [hint, setHint] = useState<string>('')

  const handleClickShowPassword = () => {
    setShowPassword(!showPassword)
  }

  const handleMouseDownPassword = (
    event: React.MouseEvent<HTMLButtonElement>
  ) => {
    event.preventDefault()
  }

  let delay: any = null
  const validate = (password: string): any => {
    if (delay !== null) {
      clearTimeout(delay)
    }
    delay = setTimeout(() => {
      const isValid = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,32}$/.test(password)
      setValid(isValid)
      setHint(password.length > 0 ? 'Invalid password.' : 'Enter password.')
      setPassword(isValid ? password : '')
      delay = null
    }, 200)
  }

  const hintString = valid ? null : (
    <FormHelperText error={!valid} id='standard-weight-helper-text'>
      {hint}
    </FormHelperText>
  )

  return (
    <FormControl
      required
      variant='outlined'
      fullWidth
      className={classes.textfield}>
      <InputLabel htmlFor='outlined-adornment-password' error={!valid}>
        Password
      </InputLabel>
      <OutlinedInput
        required
        error={!valid}
        id='outlined-adornment-password'
        type={showPassword ? 'text' : 'password'}
        onChange={e => validate(e.target.value)}
        endAdornment={
          <InputAdornment position='end'>
            <IconButton
              aria-label='toggle password visibility'
              onClick={handleClickShowPassword}
              onMouseDown={handleMouseDownPassword}>
              {showPassword ? <Visibility /> : <VisibilityOff />}
            </IconButton>
          </InputAdornment>
        }
        labelWidth={80}
      />
      <FormHelperText error={!valid} id='standard-weight-helper-text'>
        {hintString}
      </FormHelperText>
    </FormControl>
  )
}

export default AuthPasswordField
