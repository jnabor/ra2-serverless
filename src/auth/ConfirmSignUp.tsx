import React, { useState, useContext, useEffect } from 'react'
import { useHistory } from 'react-router-dom'
import { Link, Grid } from '@material-ui/core'

import Layout from '../app/AppLayout'
import Snackbar from '../common/Snackbar'

import { AuthContext } from './auth-context'
import AuthCodeField from './components/AuthCodeField'
import AuthButton from './components/AuthButton'
import AuthLayout from './components/AuthLayout'
import { useStyles } from './components/styles'

export interface AuthConfirmSignUpProps {}

const AuthConfirmSignUp: React.SFC<AuthConfirmSignUpProps> = () => {
  const authContext = useContext(AuthContext)
  const [code, setCode] = useState<string>('')
  const [disable, setDisable] = useState<boolean>(true)
  const [error, setError] = useState<string>('')
  const [message, setMessage] = useState<string>('')
  const history = useHistory()

  useEffect(() => {
    setDisable(!code)
  }, [code])

  const submitHandler = (e: any) => {
    e.preventDefault()
    authContext
      .confirmSignUp(code)
      .then(data => {
        console.log(data)
        history.push('/')
      })
      .catch(err => {
        console.error('error:', err)
        setError(err)
      })
  }

  const resendHandler = () => {
    setMessage('')
    authContext
      .resendSignUp()
      .then(data => {
        setMessage('Code resent to your email.')
      })
      .catch(err => {
        console.error('error:', err)
        setError(err)
      })
  }

  const classes = useStyles()
  return (
    <Layout title='RA2 Confirm Sign Up'>
      <AuthLayout title='Confirm'>
        <Snackbar variant='error' message={error} />
        <Snackbar variant='success' message={message} />
        <form
          className={classes.form}
          onSubmit={e => submitHandler(e)}
          noValidate>
          <AuthCodeField setCode={code => setCode(code)} />
          <AuthButton disabled={disable}>Confirm</AuthButton>
          <Grid container>
            <Grid item xs className={classes.links}></Grid>
            <Grid item className={classes.links}>
              <Link href='#' onClick={() => resendHandler()} variant='body2'>
                Resend code
              </Link>
            </Grid>
          </Grid>
        </form>
      </AuthLayout>
    </Layout>
  )
}

export default AuthConfirmSignUp
