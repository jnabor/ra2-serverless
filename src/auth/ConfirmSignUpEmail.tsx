import React, { useState, useContext, useEffect } from 'react'
import { useHistory } from 'react-router-dom'
import { Link, Grid, TextField } from '@material-ui/core'

import Layout from '../app/AppLayout'
import Snackbar from '../common/Snackbar'

import { AuthContext } from './auth-context'
import AuthCodeField from './components/AuthCodeField'
import AuthEmailField from './components/AuthEmailField'
import AuthButton from './components/AuthButton'
import AuthLayout from './components/AuthLayout'
import { useStyles } from './components/styles'

export interface AuthConfirmSignUpProps {}

const AuthConfirmSignUp: React.SFC<AuthConfirmSignUpProps> = () => {
  const authContext = useContext(AuthContext)
  const [email, setEmail] = useState<string>('')
  const [code, setCode] = useState<string>('')
  const [disable, setDisable] = useState<boolean>(true)
  const [error, setError] = useState<string>('')
  const [message, setMessage] = useState<string>('')
  const history = useHistory()

  useEffect(() => {
    setDisable(!(email && code))
  }, [email, code])

  const submitHandler = (e: any) => {
    e.preventDefault()
    console.log('resend confirmation code sign up', code)
  }

  const resendHandler = () => {
    setMessage('')
    authContext
      .resendSignUp()
      .then(data => {
        setMessage('Code resent to your email.')
      })
      .catch(err => {
        console.error('error:', err)
        setError(err)
      })
  }

  const classes = useStyles()
  return (
    <Layout title='RA2 Confirm Sign Up'>
      <AuthLayout title='Confirm'>
        <Snackbar variant='error' message={error} />
        <form
          className={classes.form}
          onSubmit={e => submitHandler(e)}
          noValidate>
          <AuthEmailField setEmail={email => setEmail(email)} />
          <AuthCodeField setCode={code => setCode(code)} />
          <AuthButton disabled={disable}>Confirm</AuthButton>
          <Grid container>
            <Grid item xs className={classes.links}></Grid>
            <Grid item className={classes.links}>
              <Link href='#' onClick={() => resendHandler()} variant='body2'>
                Resend code
              </Link>
            </Grid>
          </Grid>
        </form>
      </AuthLayout>
    </Layout>
  )
}

export default AuthConfirmSignUp
