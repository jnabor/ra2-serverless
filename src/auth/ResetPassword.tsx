import React, { useState } from 'react'
import { useHistory } from 'react-router-dom'
import { Link, Grid } from '@material-ui/core'

import Layout from '../app/AppLayout'
import AuthButton from './components/AuthButton'
import AuthEmailField from './components/AuthEmailField'
import AuthLayout from './components/AuthLayout'
import { useStyles } from './components/styles'

export interface AuthResetProps {}

const AuthReset: React.SFC<AuthResetProps> = () => {
  const [email, setEmail] = useState<string>('')
  const history = useHistory()

  const submitHandler = (e: any) => {
    e.preventDefault()
    console.log('reset password')
  }

  const classes = useStyles()

  return (
    <Layout title='RA2 Reset Password'>
      <AuthLayout title='Reset'>
        <form
          className={classes.form}
          onSubmit={e => submitHandler(e)}
          noValidate>
          <AuthEmailField setEmail={email => setEmail(email)} />
          <AuthButton disabled={true}>Reset</AuthButton>
        </form>
      </AuthLayout>
    </Layout>
  )
}

export default AuthReset
