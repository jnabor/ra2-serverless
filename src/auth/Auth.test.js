import React from 'react'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import SignIn from './SignIn'
import SignUp from './SignUp'
import ConfirmSignUp from './ConfirmSignUp'
import ConfirmSignUpEmail from './ConfirmSignUpEmail'
import ResetPassword from './ResetPassword'

import AuthLayout from './components/AuthLayout'
import AuthButton from './components/AuthButton'
import AuthEmailField from './components/AuthEmailField'
import AuthPasswordField from './components/AuthPasswordField'
import AuthCodeField from './components/AuthCodeField'

configure({ adapter: new Adapter() })

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn()
  })
}))

describe('<SignIn />', () => {
  const container = shallow(<SignIn />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<SignUp />', () => {
  const container = shallow(<SignUp />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<ConfirmSignUp />', () => {
  const container = shallow(<ConfirmSignUp />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<ConfirmSignUpEmail />', () => {
  const container = shallow(<ConfirmSignUpEmail />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<ResetPassword />', () => {
  const container = shallow(<ResetPassword />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<AuthLayout />', () => {
  const container = shallow(<AuthLayout />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<AuthButton></AuthButton>', () => {
  const container = mount(<AuthButton>Test</AuthButton>)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<AuthEmailField />', () => {
  const container = mount(<AuthEmailField />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<AuthPasswordField />', () => {
  const container = mount(<AuthPasswordField />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})

describe('<AuthCodeField />', () => {
  const container = mount(<AuthCodeField />)
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })
})
