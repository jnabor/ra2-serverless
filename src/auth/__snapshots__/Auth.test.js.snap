// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthButton></AuthButton> should match the snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-5 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Test</span><span class=\\"MuiTouchRipple-root\\"></span></button>"`;

exports[`<AuthCodeField /> should match the snapshot 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"confirmation\\" id=\\"confirmation-label\\">Confirmation Code<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"confirmation\\" name=\\"confirmation\\" required=\\"\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"PrivateNotchedOutline-root-108 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legend-109\\" style=\\"width: 0.01px;\\"><span>​</span></legend></fieldset></div></div>"`;

exports[`<AuthEmailField /> should match the snapshot 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"email\\" id=\\"email-label\\">Email Address<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"email\\" id=\\"email\\" name=\\"email\\" required=\\"\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"PrivateNotchedOutline-root-108 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legend-109\\" style=\\"width: 0.01px;\\"><span>​</span></legend></fieldset></div></div>"`;

exports[`<AuthLayout /> should match the snapshot 1`] = `"<div class=\\"MuiContainer-root MuiContainer-maxWidthXs\\"><div class=\\"makeStyles-paper-3\\"><div class=\\"MuiAvatar-root MuiAvatar-circle makeStyles-avatar-4 MuiAvatar-colorDefault\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><g fill=\\"none\\"><path d=\\"M0 0h24v24H0V0z\\"></path><path d=\\"M0 0h24v24H0V0z\\" opacity=\\".87\\"></path></g><path d=\\"M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z\\"></path></svg></div><h1 class=\\"MuiTypography-root MuiTypography-h5\\"></h1></div></div>"`;

exports[`<AuthPasswordField /> should match the snapshot 1`] = `"<div class=\\"MuiFormControl-root makeStyles-textfield-2 MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"outlined-adornment-password\\">Password<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd\\" required=\\"\\"><input aria-invalid=\\"false\\" id=\\"outlined-adornment-password\\" required=\\"\\" type=\\"password\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd\\" value=\\"\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"toggle password visibility\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"m0 0h24v24H0z\\"></path><path d=\\"M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 42px;\\" class=\\"PrivateNotchedOutline-root-108 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legend-109\\" style=\\"width: 0.01px;\\"><span>​</span></legend></fieldset></div></div>"`;

exports[`<ConfirmSignUp /> should match the snapshot 1`] = `""`;

exports[`<ConfirmSignUpEmail /> should match the snapshot 1`] = `""`;

exports[`<ResetPassword /> should match the snapshot 1`] = `""`;

exports[`<SignIn /> should match the snapshot 1`] = `""`;

exports[`<SignUp /> should match the snapshot 1`] = `""`;
